src/realm/object-store/shared_realm.hpp

```diff
@@ -546,7 +550,12 @@ class Realm : public std::enable_shared_from_this<Realm> {
 
     // Ensure that m_schema and m_schema_version match that of the current
     // version of the file
-    void read_schema_from_group_if_needed();
+    void read_schema_from_group_if_needed(bool is_schema_additive = true);
+
+    // check if current transaction version is lower than last transaction version for the latest cached schema.
+    // In case this is happening and we have opened realm in additive mode, we load the cached schema.
+    // This could happed when new properties and classes are added to the schema if realm is open in additive mode.
```

Comment:
```suggestion
    // Check if the current transaction version is lower than the last transaction version for the latest cached schema.
    // If this happens and we have opened the Realm in additive mode, we load the cached schema.
    // This can happen when new properties and classes are added to the schema if the Realm is open in additive mode.
```